ubuntu-advantage

```diff
@@ -57,6 +57,16 @@ check_result() {
     fi
 }
 
+http_request_code() {
+    curl -ILs "$1" | awk '/^HTTP\// { code=$2; }; END { print code; }'
```

Comment:
We should check if we can add curl as a dependency of ubuntu-advantage-tools. If yes, then we need to update debian/control. We are already pulling in the gnutls library because of apt-transport-https, so if we go ahead with this then we should make sure we are using the gnutls version of libcurl3

Reply:
They don't seem to be alternatives. apt-transport-https depends on libcurl3-gnutls, whereas curl depends on libcurl3.
FWIW I don't see a reason we can't depend on curl.

Reply:
After a discussion in IRC (See https://irclogs.ubuntu.com/2017/11/27/%23ubuntu-devel.html#t11:34), we decided to abandon curl as it indeed pulls in too much for ubuntu-minimal. It was suggested that we try these:
 * from trusty onwards: `/usr/lib/apt/apt-helper download-file <url> <path>` (not available in precise)
 * or talk to the https method directly, like shown in http://paste.ubuntu.com/26057904/
    $ /usr/lib/apt/methods/http
    100 Capabilities
    Version: 1.2
    Pipeline: true
    Send-Config: true
    
    600 URI Acquire
    URI: https://jak-linux.org
    FileName: /tmp/test

Reply:
I'm fine with skipping this auth-check feature for precise. Talking to the https method over stdin/stdout/stderr via shell seems icky.